### DISCUSSION TITLE
Website with outstatic crash only on vercel

### DISCUSSION THREAD

**Discussion by FrancocDev** [vercel/community#6918] [Help]
### Summary

Hi, I add Outstatic to my portfolio (CMS file based) and crash only on vercel. Works well with "next dev", "vercel dev" and the builds didn't fail in local and in vercel. 

The website retrieves this: 
`Application error: a server-side exception has occurred (see the server logs for more information).
Digest: 2192603207`

And the logs show this:
`[GET] /en?nxtPlang=en status=500

[Error: ENOENT: no such file or directory, open '/var/task/outstatic/content/metadata.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/var/task/outstatic/content/metadata.json'
}

[Error: An error occurred in the Server Components render. The specific message is omitted in production builds to avoid leaking sensitive details. A digest property is included on this error instance which may provide additional details about the nature of the error.] {
  digest: '2192603207'
}
`

For context, I'm using a custom middleware to set up a cookie based on the lang and then, passed it to outstatic to filter the data. After add outstatic, I use it with a plain json and just works. 
On the outstatic repo, the maintener said that is a edge network problem.

### Steps to Reproduce

Add oustatic to a next.js project, pass a dynamic argument to filter the data, deploy to vercel.

**Comment by versecafe** [comment]
@FrancocDev Could you share your repo, or a minimal reproduction, I'm guessing this is something to do with the serverless runtime and writing/reading from the fs, but I can't be certain without more information.

**Reply by FrancocDev** [reply]
Sorry for the delay. 
Here is a minimal reproduction: https://github.com/FrancocDev/demo-outstatic
Here is the deploy: https://demo-outstatic-pr14b5sbw-francocdev.vercel.app
